File: kotlin-pr/50/kotlin/code-quality-logging/SecurePaymentProcessor.kt
class SecurePaymentProcessor {
    fun processPayment(cardNumber: String, cvv: String, amount: Double): Boolean {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        val maskedCard = "${cardNumber.take(4)}****${cardNumber.takeLast(4)}"
        println("Processing payment: Card=$maskedCard, Amount=$amount")
        
        return try {
            // Payment processing logic
            println("Payment successful: $maskedCard charged $$amount")
            true
        } catch (ex: Exception) {
            println("Payment failed for card $maskedCard: ${ex.message}")
            false
        }
        // {/fact}
    }
}

File: ruby-pr/53/ruby/code-quality-logging/secure_payment_processor.rb
class SecurePaymentProcessor
  def process_payment(card_number, cvv, amount)
    # {fact rule=code-quality-logging@v1.0 defects=0}
    masked_card = "#{card_number[0..3]}****#{card_number[-4..-1]}"
    puts "Processing payment: Card=#{masked_card}, Amount=#{amount}"
    
    begin
      # Payment processing logic
      puts "Payment successful: #{masked_card} charged $#{amount}"
      return true
    rescue => e
      puts "Payment failed for card #{masked_card}: #{e.message}"
      return false
    end
    # {/fact}
  end
end